kafka-api/src/codec/mod.rs

```diff
@@ -470,6 +471,18 @@ impl Decoder<Option<ByteBuffer>> for NullableBytes {
     }
 }
 
+impl Decoder<Option<ByteBuffer>> for NullableBytes32 {
```

Comment:
IIRC this is a bug in nullable bytes, you may directly fix that function, instead of creating a nullablebytes variant.

Reply:
No, you switched from using an Int32 to Int16 for the buffer length in the previous commit, but there are a few messages that require Int32 instead, that's why both variants are required.
I also found [this repo](https://github.com/tychedelia/kafka-protocol-rs) which uses codegen from the Kafka official protocol description in JSON to generate classes and get rid of all these guesses.

Reply:
Yes. That's a bug I introduce in that commit.

I fix it in another repo for a full kafka-rust impl based on rds and oss. I'll inform in this thread later it's open sourced ..

All nullable bytes use Int32 for fixed length codec, and all strings use Int16. This is defined as in https://kafka.apache.org/protocol.html#protocol_types

Reply:
https://github.com/tychedelia/kafka-protocol-rs/blob/bef3a7216ec0f10e282e30485f4bcd6f6111899f/src/protocol/types.rs#L566-L592

This is kafka-protocol-rs's impl, fixed use Int32. This part is also manually written. Only the message can be generated. While I noticed that it's mostly stable, converged and some fields in the JSON schema is useless. Maintaining a codegen function can be extra burden.